pinot-spi/src/main/java/org/apache/pinot/spi/utils/GroovyTemplateUtils.java

```diff
@@ -33,15 +34,18 @@ public class GroovyTemplateUtils {
   private GroovyTemplateUtils() {
   }
 
-  private static final SimpleTemplateEngine GROOVY_TEMPLATE_ENGINE = new SimpleTemplateEngine();
+  private static final GroovyShell GROOVY_SHELL = new GroovyShell();
+  private static final SimpleTemplateEngine GROOVY_TEMPLATE_ENGINE = new SimpleTemplateEngine(GROOVY_SHELL);
   private static final DateTimeFormatter DATE_FORMAT =
       DateTimeFormatter.ofPattern("yyyy-MM-dd").withZone(ZoneOffset.UTC);
 
   public static String renderTemplate(String template, Map<String, Object> newContext)
       throws IOException, ClassNotFoundException {
     Map<String, Object> contextMap = getDefaultContextMap();
     contextMap.putAll(newContext);
-    return GROOVY_TEMPLATE_ENGINE.createTemplate(template).make(contextMap).toString();
+    String templateRendered = GROOVY_TEMPLATE_ENGINE.createTemplate(template).make(contextMap).toString();
+    GROOVY_SHELL.getClassLoader().clearCache();
```

Comment:
this means the cache will not kick in if we try to re-register the same template. but i don't think this will be created on a per-row basis